src/NumberGuessingGame.jsx

```diff
@@ -11,67 +11,56 @@ function getRandomNumber() {
   return Math.floor(Math.random() * 100) + 1;
 }
 
-const MAX_ATTEMPTS = 5;
+const MAX_ATTEMPTS = 8;
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
+function NumberGuessingGame() {
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
+  
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0);
+  const [latestGuess, setLatestGuess] = useState(null);
+  
+  const isCorrectGuess = latestGuess === numberToGuess;
+  const isGameOver =
+    isCorrectGuess || numberOfGuesses === MAX_ATTEMPTS;
+  
+  const handleGuess = (guess) => {
+    setLatestGuess(Number(guess));
+    setNumberToGuess(prev => prev + 1);
```

Comment:
Lol, is this a troll?  I thought there was a bug somewhere because it seems like the game is impossible